core/crypto/sm3.c

```diff
@@ -0,0 +1,290 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2019 Huawei Technologies Co., Ltd
+ */
+/*
+ * SM3 Hash algorithm
+ * thanks to Xyssl
+ * author:goldboar
+ * email:goldboar@163.com
+ * 2011-10-26
+ */
+
+#include <string.h>
+#include <string_ext.h>
+
+#include "sm3.h"
+
+#define GET_UINT32_BE(n, b, i)			\
```

Comment:
sorry, nitpicking.
For consistency with other implementation, prefer the 1st form I proposed: with `do { ... } while (0)` around.

```C
#define GET_UINT32_BE(n, b, i)				\
	do {						\
		(n) = ((uint32_t)(b)[(i)] << 24)     |	\
		      ((uint32_t)(b)[(i) + 1] << 16) |	\
		      ((uint32_t)(b)[(i) + 2] <<  8) |	\
		      ((uint32_t)(b)[(i) + 3]);		\
	} while (0)

Reply:
OK will change and same for `PUT_UINT32_BE()`.